components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3167,9 +3301,27 @@ public void updateIdPWithResourceId(String resourceId, IdentityProvider
                 }
                 List<IdentityProviderProperty> identityProviderProperties = getCombinedProperties
                         (newIdentityProvider.getJustInTimeProvisioningConfig(), idpProperties);
+
+                boolean hasTrustedJwtIssuerProperty = false;
+                for (IdentityProviderProperty property : identityProviderProperties) {
+                    if (property.getName().equals(IS_TRUSTED_JWT_ISSUER)) {
+                        property.setValue(String.valueOf(newIdentityProvider.getIsTrustedJwtIssuer()));
+                        hasTrustedJwtIssuerProperty = true;
+                        break;
+                    }
+                }
+
+                if (!hasTrustedJwtIssuerProperty) {
```

Comment:
if the property is present but the value has changed don't we still need to update it?